lib/service/task_service.dart

```diff
@@ -20,27 +21,60 @@ class TaskService {
     }
   }
 
-  Future<List<Task>> readTasks() async {
-    dynamic data = await client.getData(uri: taskApiBase);
-    List<Task> tasks =
-        List<Task>.from(data.map((taskData) => Task.fromJson(taskData)));
-    print(data);
-    return tasks;
-  }
-
-  Future<void> editTask({required Task task}) async {
+  Future<List<Task>> readTasksByStatus(TaskStatus status) async {
     try {
-      await client.putData(
-        uri: '$taskApiBase/${task.id}',
-        body: task.toJson(),
+      dynamic data = await client.getData(
+        uri: '/tasks',
+        queryParams: {'status': Task.getStatusString(status)},
       );
+
+      if (data is List) {
+        List<Task> tasks =
+            List<Task>.from(data.map((taskData) => Task.fromJson(taskData)));
+        return tasks;
```

Comment:
Please change this to:

```suggestion
        return List<Task>.from(data.map((taskData) => Task.fromJson(taskData)));
```

Reply:
done